interface/modules/custom_modules/oe-module-faxsms/src/Controller/AppDispatch.php

```diff
@@ -0,0 +1,537 @@
+     */
+    abstract function sendSMS(): mixed;
+
+    /**
+     * @return string|bool
+     */
+    abstract function fetchReminderCount(): string|bool;
+
+    /**
+     * @param $param
+     * @param $default
+     * @return mixed|null
+     */
+    public function getPost($param = null, $default = null): mixed
+    {
+        if ($param) {
+            return $this->_post[$param] ?? $default;
+        }
+
+        return $this->_post;
+    }
+
+    /**
+     * @param $param
+     * @param $default
+     * @return mixed|null
+     */
+    public function getServer($param = null, $default = null): mixed
+    {
+        if ($param) {
+            return $this->_server[$param] ?? $default;
+        }
+
+        return $this->_server;
+    }
+
+    /**
+     * @param $key
+     * @param $value
+     * @return $this
+     */
+    public function setSession($key, $value): static
+    {
+        // ensure write is allowed by using utility.
+        SessionUtil::setSession($key, $value);
+        return $this;
+    }
+
+    /**
+     * @param array $setup
+     * @return string
+     */
+    protected function saveSetup(array $setup = []): string
+    {
+        if (empty($setup)) {
+            $username = $this->getRequest('username');
+            $ext = $this->getRequest('extension');
+            $account = $this->getRequest('account');
+            $phone = $this->getRequest('phone');
+            $password = $this->getRequest('password');
+            $appkey = $this->getRequest('key');
+            $appsecret = $this->getRequest('secret');
+            $production = $this->getRequest('production');
+            $smsNumber = $this->getRequest('smsnumber');
+            $smsMessage = $this->getRequest('smsmessage');
+            $smsHours = $this->getRequest('smshours');
+            $setup = array(
+                'username' => "$username",
+                'extension' => "$ext",
+                'account' => $account,
+                'phone' => $phone,
+                'password' => "$password",
+                'appKey' => "$appkey",
+                'appSecret' => "$appsecret",
+                'server' => !$production ? 'https://platform.devtest.ringcentral.com' : "https://platform.ringcentral.com",
+                'portal' => !$production ? "https://service.devtest.ringcentral.com/" : "https://service.ringcentral.com/",
+                'smsNumber' => "$smsNumber",
+                'production' => $production,
+                'redirect_url' => $this->getRequest('redirect_url') ?: $_REQUEST['redirect_url'],
+                'smsHours' => $smsHours,
+                'smsMessage' => $smsMessage
+            );
+        }
+
+        $vendor = self::getModuleVendor();
+        $this->authUser = (int)$this->getSession('authUserID') ?? 0;
+        if (!($GLOBALS['oerestrict_users'] ?? null)) {
+            $this->authUser = 0;
+        }
+        // encrypt for safety.
+        $content = $this->crypto->encryptStandard(json_encode($setup));
```

Comment:
this is correct use of encryptStandard :) (since using drive key to encrypt in database)